File: core/src/main/scala/com/alibaba/datax/core/transport/record/TerminateRecord.scala
package com.alibaba.datax.core.transport.record

import com.alibaba.datax.common.scala.element.{Column, Record}

//import com.alibaba.datax.common.element.Column
//import com.alibaba.datax.common.element.Record

/**
 * 作为标示 生产者已经完成生产的标志
 *
 */
object TerminateRecord {
  private val SINGLE = new TerminateRecord

  def get = SINGLE
}

class TerminateRecord private() extends Record {
  override def addColumn(column: Column): Unit = ???

  override def setColumn(i: Integer, column: Column): Unit = ???

  override def getColumn(i: Integer): Column = ???

  override def getColumnNumber: Integer = ???

  override def getByteSize: Integer = ???

  override def getMemorySize: Integer = ???
}

File: core/src/main/java/com/alibaba/datax/core/transport/record/TerminateRecord.java
package com.alibaba.datax.core.transport.record;

import com.alibaba.datax.common.element.Column;
import com.alibaba.datax.common.element.ICol2Index;
import com.alibaba.datax.common.element.Record;
import com.alibaba.datax.plugin.rdbms.reader.util.DataXCol2Index;

import java.util.Map;

/**
 * 作为标示 生产者已经完成生产的标志
 */
public class TerminateRecord implements Record {
    private final static TerminateRecord SINGLE = new TerminateRecord();

    private TerminateRecord() {
    }

    @Override
    public void setCol2Index(ICol2Index mapper) {

    }

    @Override
    public String getString(String field, boolean origin) {
        throw new UnsupportedOperationException();
    }

    @Override
    public ICol2Index getCol2Index() {
        throw new UnsupportedOperationException();
    }

    public static TerminateRecord get() {
        return SINGLE;
    }

    @Override
    public void addColumn(Column column) {
    }

    @Override
    public void setString(String field, String val) {

    }

    @Override
    public void setColumn(String field, Object column) {

    }

    @Override
    public Object getColumn(String field) {
        return null;
    }

    @Override
    public Column getColumn(int i) {
        return null;
    }

    @Override
    public int getColumnNumber() {
        return 0;
    }

    @Override
    public int getByteSize() {
        return 0;
    }

    @Override
    public int getMemorySize() {
        return 0;
    }

    @Override
    public void setColumn(int i, Column column) {
        return;
    }
}
